src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -101,10 +156,23 @@ public void open(Collection<TopicPartition> partitions) {
                     // The offset was not found, so rather than forcibly set the offset to 0 we let the
                     // consumer decide where to start based upon standard consumer offsets (if available)
                     // or the consumer's `auto.offset.reset` configuration
+
+                    // if we are in snapshot mode, then just start at zero.
+                    if (config.snapShotMode()) {
+                        context.offset(topicPartition, 0);
+                    }
+
                     log.info("Resetting offset for {} based upon existing consumer group offsets or, if "
                             + "there are none, the consumer's 'auto.offset.reset' value.", topicPartition);
                 }
+
                 this.partitionWriters.put(topicPartition, partitionWriter);
+                this.currentOffsets.put(topicPartition, lastWrittenOffset);
+                log.debug("last written offset " + lastWrittenOffset);
```

Comment:
we should either print the topicpartition here or remove the message

Reply:
Removed